### PR TITLE
Add ExplosiveProjectile

### PR DESCRIPTION
**Pull request by Aaron1011**
This PR creates a new `ExplosiveProjectile` interface, which represents projectiles such as `WitherSkull`s and `Fireball`s which create an explosion in addition to functioning as `Projectile`s.

Following the discussion in issue https://github.com/SpongePowered/SpongeAPI/issues/339, I've decided not to create a separation between block damaging and non block damaging projectiles.

This resolves https://github.com/SpongePowered/SpongeAPI/issues/339

### CODE REVIEW

**Comment by ST-DDT** [comment]
Why is the `SmallFireball`  in explosives package?

**Comment by Aaron1011** [comment]
@ST-DDT: It still creates an explosion, although it's not block damaging. See my explanation in the description, as well as the linked issue.

File: src/main/java/org/spongepowered/api/entity/projectile/explosive/ExplosiveProjectile.java [hunk 1]
```diff
@@ -22,18 +22,21 @@
...
  */
+package org.spongepowered.api.entity.projectile.explosive;
 
-package org.spongepowered.api.entity.projectile.fireball;
+import org.spongepowered.api.entity.explosive.Explosive;
+import org.spongepowered.api.entity.projectile.DamagingProjectile;
 
 /**
- * Represents a Ghast fireball.
+ * Represents an {@link DamagingProjectile} which is also an {@link Explosive}
```
Comment (me4502) [inline src/main/java/org/spongepowered/api/entity/projectile/explosive/ExplosiveProjectile.java]: Represents a*

File: src/main/java/org/spongepowered/api/entity/projectile/explosive/ExplosiveProjectile.java [hunk 2]
```diff
@@ -22,18 +22,21 @@
...
  */
+package org.spongepowered.api.entity.projectile.explosive;
 
-package org.spongepowered.api.entity.projectile.fireball;
+import org.spongepowered.api.entity.explosive.Explosive;
+import org.spongepowered.api.entity.projectile.DamagingProjectile;
 
 /**
- * Represents a Ghast fireball.
+ * Represents a {@link DamagingProjectile} which is also an {@link Explosive}
```
Comment (simon816) [inline src/main/java/org/spongepowered/api/entity/projectile/explosive/ExplosiveProjectile.java]: Missing full-stop

**Comment by gratimax** [comment]
63da58927e4762e678c

**Pull request closed without merging**